### ISSUE TITLE
Add support for inline steps

### ISSUE THREAD

**Issue by Suremaker** [LightBDD/LightBDD#156]
I'm just getting started with LightBDD and that's one of my only objections so far. I'm moving all my unit tests to a BDD structure with either class per story or class per scenario, and going from classic xUnit to Light BDD adds additional verbosity when a step isn't shared (which is most of the time) and could be inlined.

BDDfy does this but they got it backwards - the first parameter is the function and the second parameter is the step title, which reduces readability big time.

SubSpec defines G/W/T (or C/D/A in their parlance) extension methods for strings.

_Originally posted by @dday4thedeceiver in https://github.com/LightBDD/LightBDD/issues/152#issuecomment-437176684_

**Comment by Suremaker** [comment]
Hello @dday4thedeceiver,

As promised, I am coming back to you with the inline steps.
While, by default, the LightBDD does not support them, it is relatively easy to add them to the test project.

Let's assume I have a following calculator class:
```c#
class Calculator
{
    public int Add(int x, int y) => x + y;
}
```

I can write the following test:
```c#
public class Calculator_feature : FeatureFixture
{
    [Scenario]
    public async Task Adding_numbers()
    {
        Calculator calc = null;
        int result = 0;

        await Runner.NewScenario()
            .AddStep("Given calculator", () => calc = new Calculator())
            .AddStep("When I add 2 numbers", () => result = calc.Add(3, 5))
            .AddStep("Then I should receive proper result", () => Assert.Equal(8, result))
            .RunAsync();
    }
}
```

... by adding this helper class with `AddStep()` method:
```c#
public static class TextStepExtensions
{
    // non-contextual, synchronous step
    public static IScenarioBuilder<T> AddStep<T>(this IScenarioBuilder<T> builder, string stepName, Action action)
    {
        builder.Integrate().AddSteps(new[]
        {
            new StepDescriptor(stepName, (_, __) =>
            {
                action();
                return Task.FromResult(DefaultStepResultDescriptor.Instance);
            })
        });
        return builder;
    }
}
```

Please let me know if that is something that you are looking for... If so, I will consider adding it instead of #155 as it appears that support for local function would require manual parsing of the unofficial signatures, as `Some_step()` local function is compiled to name like `<GetStepName_should_interpret_local_functions>g__Some_step|1_0` and there is no official api to receive the original name of it.

**Comment by dday4thedeceiver** [comment]
That would work... although would be nice see a single character wrapper
for AddStep as well (s perhaps?)

**Comment by Suremaker** [comment]
Hello, the change is implemented in LightBDD 2.5.0.

I have added the documentation on wiki page: https://github.com/LightBDD/LightBDD/wiki/Scenario-Steps-Definition#compact-scenarios

The example can be found here (all integrations have it present): [Compact_calculator_scenarios.cs](https://github.com/LightBDD/LightBDD/blob/master/examples/Example.LightBDD.NUnit3/Features/Compact_calculator_scenarios.cs).

Regarding the single character wrapped, I decided to not adding it as fully named methods are easier to understand and use. Feel free however to add it in your project as an extension method if you find it useful.
